File: gunhoflash/combination-sum-iv.py
from typing import List

class Solution:
  def combinationSum4(self, nums: List[int], target: int) -> int:
    answers = [0] * (target + 1)
    answers[0] = 1

    for i in range(target + 1):
      for num in nums:
        if num <= i:
          answers[i] += answers[i - num]

    return answers[target]

File: gunhoflash/combination-sum-iv.ts
function combinationSum4(nums: number[], target: number): number {
  const answers = Array(target + 1).fill(0);
  answers[0] = 1;

  for (let i = 1; i <= target; i++) {
    for (const num of nums) {
      if (num <= i) {
        answers[i] += answers[i - num];
      }
    }
  }

  return answers[target];
}
